test/test_pipeline/test_regression.py

```diff
@@ -169,8 +169,29 @@ def _test_configurations(self, configurations_space, make_sparse=False,
             cls = SimpleRegressionPipeline(random_state=1,
                                            dataset_properties=dataset_properties)
             cls.set_hyperparameters(config)
+
+            # First make sure that for this configuration, setting the parameters
+            # does not mistakenly set the estimator as fitted
+            for name, step in cls.named_steps.items():
+                with self.assertRaisesRegex(sklearn.exceptions.NotFittedError,
+                                            "instance is not fitted yet"):
+                    check_is_fitted(step)
+
             try:
                 cls.fit(X_train, Y_train)
+                # After fit, the Regressor must be tagged as fitted
+                # by sklearn. Check is fitted raises an exception if that
+                # is not the case
+                # This is not enforced over every step of the pipeline, as
```

Comment:
Would it make sense to enforce this to make this test stricter?

Reply:
Yeah we can, adding something like:
https://github.com/scikit-learn/scikit-learn/blob/c3f25168fe8c73b82af8f57ae536f4a9708c863e/sklearn/metrics/_plot/tests/test_plot_precision_recall.py#L67

The only reason I did not do it is that it will require changing couple of places here and there, but if you think it makes sense I will go with the change.  Definitely it is a nice testing mechanism to ensure that the components were actually fitted. 

Would add a new push for this.

Reply:
> but if you think it makes sense I will go with the change

I'd appreciate that